pkg/api/pod/util.go

```diff
@@ -474,6 +485,19 @@ func shareProcessNamespaceInUse(podSpec *api.PodSpec) bool {
 	return false
 }
 
+func tokenRequestProjectionInUse(podSpec *api.PodSpec) bool {
+	if podSpec == nil {
+		return false
+	}
+        for i := range podSpec.Volumes {
+		if podSpec.Volumes[i].Projected != nil &&
+			podSpec.Volumes[i].Projected.Sources != nil && podSpec.Volumes[i].Projected.Sources[i].ServiceAccountToken != nil {
```

Comment:
`i` is not a valid index into `Sources`... might be simpler to range over objects like this:
```
for _, v := range podSpec.Volumes {
  if v.Projected == nil {
    continue
  }
  for _, s := range v.Projected.Sources {
    if s.ServiceAccountToken != nil {
      return true
    }
  }
}
```